enterprise/core-edge/src/main/java/org/neo4j/commandline/dbms/UnbindFromClusterCommand.java

```diff
@@ -0,0 +1,184 @@
+    {
+        this.homeDir = homeDir;
+        this.fsa = fsa;
+    }
+
+    @Override
+    public void execute( String[] args ) throws IncorrectUsage, CommandFailed
+    {
+        Args parsedArgs = Args.parse( args );
+
+        try
+        {
+            Path pathToSpecificDatabase =
+                    homeDir.resolve( "data" ).resolve( "databases" ).resolve( databaseNameFrom( parsedArgs ) );
+            confirmTargetDatabaseDirectoryExists( pathToSpecificDatabase );
+            confirmClusterStateDirectoryExists( Paths.get( pathToSpecificDatabase.toString(), "cluster-state" ) );
+            confirmTargetDirectoryIsWritable( pathToSpecificDatabase );
+            deleteClusterStateIn( clusterStateFrom( pathToSpecificDatabase ) );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+        catch ( UnbindFailureException e )
+        {
+            throw new CommandFailed( "Unbind failed: " + e.getMessage(), e );
+        }
+    }
+
+    private Path clusterStateFrom( Path target )
+    {
+        return Paths.get( target.toString(), CLUSTER_STATE_DIRECTORY_NAME );
+    }
+
+    private void confirmTargetDirectoryIsWritable( Path dbDir ) throws CommandFailed
+    {
+        Path lockFile = dbDir.resolve( StoreLocker.STORE_LOCK_FILENAME );
+        if ( exists( lockFile ) )
+        {
+            if ( Files.isWritable( lockFile ) )
+            {
+                StoreLocker storeLocker = new StoreLocker( fsa );
+                try
+                {
+                    storeLocker.checkLock( dbDir.toFile() );
+                }
+                catch ( StoreLockException e )
+                {
+                    throw new CommandFailed( "Database is currently locked. Is a Neo4j instance still using it?" );
+                }
+            }
+        }
+    }
+
+    private void confirmTargetDatabaseDirectoryExists( Path target )
+    {
+        if ( !exists( target ) )
+        {
+            throw new IllegalArgumentException( format( "Database %s does not exist", target ) );
+        }
+    }
+
+    private void confirmClusterStateDirectoryExists( Path clusterStateDirectory ) throws UnbindFailureException
+    {
+        if ( !exists( clusterStateDirectory ) )
+        {
+            throw new UnbindFailureException( "Database %s is not bound to any cluster", clusterStateDirectory );
+        }
+    }
+
+    private void deleteClusterStateIn( Path target ) throws UnbindFailureException
+    {
+        try
+        {
+            FileUtils.deleteRecursively( target.toFile() );
+        }
+        catch ( IOException e )
+        {
+            throw new UnbindFailureException( e );
+        }
+    }
+
+    private String databaseNameFrom( Args parsedArgs )
+    {
+        String databaseName = parsedArgs.get( "database" );
+        if ( databaseName == null )
+        {
+            throw new IllegalArgumentException(
```

Comment:
We use `IncorrectUsage` in `AdminCommand`s. `CheckConsistencyCommand` is another example of this pattern.